{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Общий план задания\n",
    "1. Скачать [http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/](http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/) файлы соответствующие хромосомам 1-22.\n",
    "2. Перевести с помощью plink https://www.cog-genomics.org/plink/1.9/ из vcf в bed\\bim\\fam\n",
    "3. Оставить в данных только SNP с помощью plink.\n",
    "4. Слить полученные файлы в один набор bed\\bim\\fam\n",
    "5. Изменить fam файл, чтобы в нем была информация о популяциях. Соответствие образец-популяция можно найти на сайт 1000 геномов.\n",
    "6. Сделать новый набор данных, чтобы в нем были только европейские популяции.\n",
    "7. http://software.genetics.ucla.edu/admixture/ - с помощью этой программы сделать анализ по 5 компонентам (базовый запуск можно найти в документации).\n",
    "\n",
    "Как результат ожидается следующее:\n",
    "\n",
    "а. Текстовый файл, какие команды использовались для каждого шага (чтобы можно было воспроизвести).\n",
    "\n",
    "б.Q файл по итогам 7 шага\n",
    "\n",
    "в. Графики разложения отельных индивидов на компоненты"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 1. Скачать http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ файлы соответствующие хромосомам 1-22.\n",
    "в google colaboratory скачал:\n",
    "\n",
    "!echo http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ALL.chr{1..22}.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz.tbi | xargs -P22 -n1 wget -nv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range(1, 23):\n",
    "    !wget http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ALL.chr{str(i)}.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2. Перевести с помощью plink https://www.cog-genomics.org/plink/1.9/ из vcf в bed\\bim\\fam\n",
    "#### 3. Оставить в данных только SNP с помощью plink."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2020-02-04 20:28:49--  http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20200121.zip\n",
      "Resolving s3.amazonaws.com (s3.amazonaws.com)... 52.216.242.246\n",
      "Connecting to s3.amazonaws.com (s3.amazonaws.com)|52.216.242.246|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 8916292 (8.5M) [application/zip]\n",
      "Saving to: ‘plink_linux_x86_64_20200121.zip’\n",
      "\n",
      "plink_linux_x86_64_ 100%[===================>]   8.50M  7.53MB/s    in 1.1s    \n",
      "\n",
      "2020-02-04 20:28:51 (7.53 MB/s) - ‘plink_linux_x86_64_20200121.zip’ saved [8916292/8916292]\n",
      "\n",
      "Archive:  plink_linux_x86_64_20200121.zip\n",
      "  inflating: plink/plink             \n",
      "  inflating: plink/LICENSE           \n",
      "  inflating: plink/toy.ped           \n",
      "  inflating: plink/toy.map           \n",
      "  inflating: plink/prettify          \n"
     ]
    }
   ],
   "source": [
    "# Устанавливаем PLINK\n",
    "!wget http://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20200121.zip\n",
    "!unzip plink_linux_x86_64_20200121.zip -d plink/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr3.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr3\n",
      "  --vcf ALL.chr3.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr3-temporary.bed + ALL.chr3-temporary.bim + ALL.chr3-temporary.fam\n",
      "written.\n",
      "5832276 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr3.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999832.\n",
      "5832276 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr3.bed + ALL.chr3.bim + ALL.chr3.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr3.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr3\n",
      "  --make-bed\n",
      "  --out ALL.chr3\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "5603261 out of 5832276 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr3.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999949.\n",
      "5603261 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr3.bed + ALL.chr3.bim + ALL.chr3.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr3 хромосоме:\n",
      "4 dup3.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr3.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr3\n",
      "  --exclude dup3.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr3\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "5603261 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr3.nosex .\n",
      "--exclude: 5603253 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999949.\n",
      "5603253 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr3.bed + ALL.chr3.bim + ALL.chr3.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr4.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr4\n",
      "  --vcf ALL.chr4.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr4-temporary.bed + ALL.chr4-temporary.bim + ALL.chr4-temporary.fam\n",
      "written.\n",
      "5732585 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr4.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999819.\n",
      "5732585 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr4.bed + ALL.chr4.bim + ALL.chr4.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr4.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr4\n",
      "  --make-bed\n",
      "  --out ALL.chr4\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "5500093 out of 5732585 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr4.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999948.\n",
      "5500093 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr4.bed + ALL.chr4.bim + ALL.chr4.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr4 хромосоме:\n",
      "9 dup4.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr4.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr4\n",
      "  --exclude dup4.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr4\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Allocated 9046 MB successfully, after larger attempt(s) failed.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "5500093 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr4.nosex .\n",
      "--exclude: 5500075 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999948.\n",
      "5500075 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr4.bed + ALL.chr4.bim + ALL.chr4.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr5.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr5\n",
      "  --vcf ALL.chr5.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Allocated 9046 MB successfully, after larger attempt(s) failed.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--vcf: ALL.chr5-temporary.bed + ALL.chr5-temporary.bim + ALL.chr5-temporary.fam\n",
      "written.\n",
      "5265763 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr5.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999838.\n",
      "5265763 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr5.bed + ALL.chr5.bim + ALL.chr5.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr5.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr5\n",
      "  --make-bed\n",
      "  --out ALL.chr5\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "5055536 out of 5265763 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr5.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999953.\n",
      "5055536 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr5.bed + ALL.chr5.bim + ALL.chr5.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr5 хромосоме:\n",
      "3 dup5.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr5.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr5\n",
      "  --exclude dup5.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr5\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "5055536 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr5.nosex .\n",
      "--exclude: 5055530 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999953.\n",
      "5055530 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr5.bed + ALL.chr5.bim + ALL.chr5.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr6.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr6\n",
      "  --vcf ALL.chr6.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr6-temporary.bed + ALL.chr6-temporary.bim + ALL.chr6-temporary.fam\n",
      "written.\n",
      "5024119 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr6.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999802.\n",
      "5024119 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr6.bed + ALL.chr6.bim + ALL.chr6.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr6.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr6\n",
      "  --make-bed\n",
      "  --out ALL.chr6\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "4816881 out of 5024119 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr6.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999935.\n",
      "4816881 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr6.bed + ALL.chr6.bim + ALL.chr6.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr6 хромосоме:\n",
      "8 dup6.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr6.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr6\n",
      "  --exclude dup6.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "4816881 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr6.nosex .\n",
      "--exclude: 4816865 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999935.\n",
      "4816865 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr6.bed + ALL.chr6.bim + ALL.chr6.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr7.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr7\n",
      "  --vcf ALL.chr7.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr7-temporary.bed + ALL.chr7-temporary.bim + ALL.chr7-temporary.fam\n",
      "written.\n",
      "4716715 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr7.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999834.\n",
      "4716715 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr7.bed + ALL.chr7.bim + ALL.chr7.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr7.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr7\n",
      "  --make-bed\n",
      "  --out ALL.chr7\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "4533180 out of 4716715 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr7.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99995.\n",
      "4533180 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr7.bed + ALL.chr7.bim + ALL.chr7.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr7 хромосоме:\n",
      "0 dup7.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr7.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr7\n",
      "  --exclude dup7.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr7\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "4533180 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr7.nosex .\n",
      "--exclude: 4533180 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99995.\n",
      "4533180 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr7.bed + ALL.chr7.bim + ALL.chr7.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr8.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr8\n",
      "  --vcf ALL.chr8.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr8-temporary.bed + ALL.chr8-temporary.bim + ALL.chr8-temporary.fam\n",
      "written.\n",
      "4597105 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr8.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999837.\n",
      "4597105 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr8.bed + ALL.chr8.bim + ALL.chr8.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr8.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr8\n",
      "  --make-bed\n",
      "  --out ALL.chr8\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "4434371 out of 4597105 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr8.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999944.\n",
      "4434371 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr8.bed + ALL.chr8.bim + ALL.chr8.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr8 хромосоме:\n",
      "30 dup8.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr8.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr8\n",
      "  --exclude dup8.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr8\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "4434371 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr8.nosex .\n",
      "--exclude: 4434311 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999944.\n",
      "4434311 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr8.bed + ALL.chr8.bim + ALL.chr8.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr9.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr9\n",
      "  --vcf ALL.chr9.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr9-temporary.bed + ALL.chr9-temporary.bim + ALL.chr9-temporary.fam\n",
      "written.\n",
      "3560687 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr9.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999838.\n",
      "3560687 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr9.bed + ALL.chr9.bim + ALL.chr9.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr9.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr9\n",
      "  --make-bed\n",
      "  --out ALL.chr9\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "3427241 out of 3560687 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr9.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999948.\n",
      "3427241 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr9.bed + ALL.chr9.bim + ALL.chr9.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr9 хромосоме:\n",
      "10 dup9.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr9.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr9\n",
      "  --exclude dup9.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr9\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "3427241 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr9.nosex .\n",
      "--exclude: 3427221 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999948.\n",
      "3427221 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr9.bed + ALL.chr9.bim + ALL.chr9.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr10.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr10\n",
      "  --vcf ALL.chr10.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr10-temporary.bed + ALL.chr10-temporary.bim +\n",
      "ALL.chr10-temporary.fam written.\n",
      "3992219 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr10.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99983.\n",
      "3992219 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr10.bed + ALL.chr10.bim + ALL.chr10.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr10.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr10\n",
      "  --make-bed\n",
      "  --out ALL.chr10\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "3837178 out of 3992219 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr10.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999945.\n",
      "3837178 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr10.bed + ALL.chr10.bim + ALL.chr10.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr10 хромосоме:\n",
      "4 dup10.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr10.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr10\n",
      "  --exclude dup10.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr10\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "3837178 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr10.nosex .\n",
      "--exclude: 3837170 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999945.\n",
      "3837170 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr10.bed + ALL.chr10.bim + ALL.chr10.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr11.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr11\n",
      "  --vcf ALL.chr11.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr11-temporary.bed + ALL.chr11-temporary.bim +\n",
      "ALL.chr11-temporary.fam written.\n",
      "4045628 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr11.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999834.\n",
      "4045628 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr11.bed + ALL.chr11.bim + ALL.chr11.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr11.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr11\n",
      "  --make-bed\n",
      "  --out ALL.chr11\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "3891530 out of 4045628 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr11.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999947.\n",
      "3891530 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr11.bed + ALL.chr11.bim + ALL.chr11.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr11 хромосоме:\n",
      "2 dup11.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr11.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr11\n",
      "  --exclude dup11.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr11\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Allocated 9046 MB successfully, after larger attempt(s) failed.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "3891530 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr11.nosex .\n",
      "--exclude: 3891526 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999947.\n",
      "3891526 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr11.bed + ALL.chr11.bim + ALL.chr11.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr12.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr12\n",
      "  --vcf ALL.chr12.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Allocated 9046 MB successfully, after larger attempt(s) failed.\n",
      "--vcf: ALL.chr12-temporary.bed + ALL.chr12-temporary.bim +\n",
      "ALL.chr12-temporary.fam written.\n",
      "3868428 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr12.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999822.\n",
      "3868428 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr12.bed + ALL.chr12.bim + ALL.chr12.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr12.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr12\n",
      "  --make-bed\n",
      "  --out ALL.chr12\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "3710299 out of 3868428 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr12.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999954.\n",
      "3710299 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr12.bed + ALL.chr12.bim + ALL.chr12.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr12 хромосоме:\n",
      "6 dup12.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr12.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr12\n",
      "  --exclude dup12.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr12\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "3710299 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr12.nosex .\n",
      "--exclude: 3710046 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999954.\n",
      "3710046 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr12.bed + ALL.chr12.bim + ALL.chr12.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr13.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr13\n",
      "  --vcf ALL.chr13.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr13-temporary.bed + ALL.chr13-temporary.bim +\n",
      "ALL.chr13-temporary.fam written.\n",
      "2857916 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr13.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999814.\n",
      "2857916 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr13.bed + ALL.chr13.bim + ALL.chr13.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr13.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr13\n",
      "  --make-bed\n",
      "  --out ALL.chr13\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2737034 out of 2857916 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr13.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999951.\n",
      "2737034 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr13.bed + ALL.chr13.bim + ALL.chr13.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr13 хромосоме:\n",
      "1 dup13.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr13.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr13\n",
      "  --exclude dup13.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr13\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2737034 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr13.nosex .\n",
      "--exclude: 2737032 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999951.\n",
      "2737032 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr13.bed + ALL.chr13.bim + ALL.chr13.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr14.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr14\n",
      "  --vcf ALL.chr14.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr14-temporary.bed + ALL.chr14-temporary.bim +\n",
      "ALL.chr14-temporary.fam written.\n",
      "2655067 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr14.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999836.\n",
      "2655067 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr14.bed + ALL.chr14.bim + ALL.chr14.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr14.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr14\n",
      "  --make-bed\n",
      "  --out ALL.chr14\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2548064 out of 2655067 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr14.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999952.\n",
      "2548064 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr14.bed + ALL.chr14.bim + ALL.chr14.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr14 хромосоме:\n",
      "2 dup14.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr14.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr14\n",
      "  --exclude dup14.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr14\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2548064 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr14.nosex .\n",
      "--exclude: 2548060 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999952.\n",
      "2548060 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr14.bed + ALL.chr14.bim + ALL.chr14.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr15.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr15\n",
      "  --vcf ALL.chr15.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr15-temporary.bed + ALL.chr15-temporary.bim +\n",
      "ALL.chr15-temporary.fam written.\n",
      "2424689 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr15.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999833.\n",
      "2424689 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr15.bed + ALL.chr15.bim + ALL.chr15.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr15.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr15\n",
      "  --make-bed\n",
      "  --out ALL.chr15\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2328557 out of 2424689 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr15.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999948.\n",
      "2328557 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr15.bed + ALL.chr15.bim + ALL.chr15.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr15 хромосоме:\n",
      "26 dup15.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr15.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr15\n",
      "  --exclude dup15.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr15\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2328557 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr15.nosex .\n",
      "--exclude: 2328505 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999948.\n",
      "2328505 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr15.bed + ALL.chr15.bim + ALL.chr15.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr16.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr16\n",
      "  --vcf ALL.chr16.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr16-temporary.bed + ALL.chr16-temporary.bim +\n",
      "ALL.chr16-temporary.fam written.\n",
      "2697949 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr16.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999847.\n",
      "2697949 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr16.bed + ALL.chr16.bim + ALL.chr16.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr16.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr16\n",
      "  --make-bed\n",
      "  --out ALL.chr16\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2607034 out of 2697949 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr16.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999947.\n",
      "2607034 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr16.bed + ALL.chr16.bim + ALL.chr16.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr16 хромосоме:\n",
      "7 dup16.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr16.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr16\n",
      "  --exclude dup16.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr16\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2607034 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr16.nosex .\n",
      "--exclude: 2607020 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999947.\n",
      "2607020 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr16.bed + ALL.chr16.bim + ALL.chr16.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr17.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr17\n",
      "  --vcf ALL.chr17.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr17-temporary.bed + ALL.chr17-temporary.bim +\n",
      "ALL.chr17-temporary.fam written.\n",
      "2329288 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr17.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999842.\n",
      "2329288 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr17.bed + ALL.chr17.bim + ALL.chr17.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr17.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr17\n",
      "  --make-bed\n",
      "  --out ALL.chr17\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2234710 out of 2329288 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr17.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999951.\n",
      "2234710 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr17.bed + ALL.chr17.bim + ALL.chr17.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr17 хромосоме:\n",
      "5 dup17.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr17.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr17\n",
      "  --exclude dup17.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr17\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2234710 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr17.nosex .\n",
      "--exclude: 2234700 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999951.\n",
      "2234700 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr17.bed + ALL.chr17.bim + ALL.chr17.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr18.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr18\n",
      "  --vcf ALL.chr18.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr18-temporary.bed + ALL.chr18-temporary.bim +\n",
      "ALL.chr18-temporary.fam written.\n",
      "2267185 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr18.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999823.\n",
      "2267185 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr18.bed + ALL.chr18.bim + ALL.chr18.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr18.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr18\n",
      "  --make-bed\n",
      "  --out ALL.chr18\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2178759 out of 2267185 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr18.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999949.\n",
      "2178759 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr18.bed + ALL.chr18.bim + ALL.chr18.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr18 хромосоме:\n",
      "2 dup18.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr18.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr18\n",
      "  --exclude dup18.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr18\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2178759 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr18.nosex .\n",
      "--exclude: 2178755 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999949.\n",
      "2178755 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr18.bed + ALL.chr18.bim + ALL.chr18.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr19.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr19\n",
      "  --vcf ALL.chr19.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr19-temporary.bed + ALL.chr19-temporary.bim +\n",
      "ALL.chr19-temporary.fam written.\n",
      "1832506 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr19.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999827.\n",
      "1832506 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr19.bed + ALL.chr19.bim + ALL.chr19.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr19.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr19\n",
      "  --make-bed\n",
      "  --out ALL.chr19\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "1758443 out of 1832506 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr19.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999942.\n",
      "1758443 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr19.bed + ALL.chr19.bim + ALL.chr19.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr19 хромосоме:\n",
      "2 dup19.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr19.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr19\n",
      "  --exclude dup19.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr19\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "1758443 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr19.nosex .\n",
      "--exclude: 1758439 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999942.\n",
      "1758439 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr19.bed + ALL.chr19.bim + ALL.chr19.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr20.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr20\n",
      "  --vcf ALL.chr20.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr20-temporary.bed + ALL.chr20-temporary.bim +\n",
      "ALL.chr20-temporary.fam written.\n",
      "1812841 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr20.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999849.\n",
      "1812841 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr20.bed + ALL.chr20.bim + ALL.chr20.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr20.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr20\n",
      "  --make-bed\n",
      "  --out ALL.chr20\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "1745171 out of 1812841 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr20.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999959.\n",
      "1745171 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr20.bed + ALL.chr20.bim + ALL.chr20.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr20 хромосоме:\n",
      "1 dup20.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr20.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr20\n",
      "  --exclude dup20.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr20\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "1745171 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr20.nosex .\n",
      "--exclude: 1745169 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999959.\n",
      "1745169 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr20.bed + ALL.chr20.bim + ALL.chr20.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr21.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr21\n",
      "  --vcf ALL.chr21.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr21-temporary.bed + ALL.chr21-temporary.bim +\n",
      "ALL.chr21-temporary.fam written.\n",
      "1105538 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr21.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999787.\n",
      "1105538 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr21.bed + ALL.chr21.bim + ALL.chr21.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr21.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr21\n",
      "  --make-bed\n",
      "  --out ALL.chr21\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "1058549 out of 1105538 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr21.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999928.\n",
      "1058549 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr21.bed + ALL.chr21.bim + ALL.chr21.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr21 хромосоме:\n",
      "1 dup21.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr21.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr21\n",
      "  --exclude dup21.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr21\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "1058549 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr21.nosex .\n",
      "--exclude: 1058547 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999928.\n",
      "1058547 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr21.bed + ALL.chr21.bim + ALL.chr21.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr22.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out ALL.chr22\n",
      "  --vcf ALL.chr22.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "--vcf: ALL.chr22-temporary.bed + ALL.chr22-temporary.bim +\n",
      "ALL.chr22-temporary.fam written.\n",
      "1103547 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr22.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99984.\n",
      "1103547 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr22.bed + ALL.chr22.bim + ALL.chr22.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr22.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr22\n",
      "  --make-bed\n",
      "  --out ALL.chr22\n",
      "  --snps-only\n",
      "  --threads 6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "1059735 out of 1103547 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr22.nosex .\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999946.\n",
      "1059735 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr22.bed + ALL.chr22.bim + ALL.chr22.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "найдено дупликатов в chr22 хромосоме:\n",
      "2 dup22.snps\n",
      " - удаляю...\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr22.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr22\n",
      "  --exclude dup22.snps\n",
      "  --make-bed\n",
      "  --out ALL.chr22\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "1059735 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr22.nosex .\n",
      "--exclude: 1059731 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999946.\n",
      "1059731 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr22.bed + ALL.chr22.bim + ALL.chr22.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "clear_output()\n",
    "for chr_num in range(1, 23):\n",
    "    #Распаковываем хромосому в bed-bim-fam файлы\n",
    "    !plink/plink --vcf ALL.chr{str(chr_num)}.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --make-bed --out ALL.chr{str(chr_num)}\n",
    "    #Оставляем только SNP\n",
    "    !plink/plink --bfile ALL.chr{str(chr_num)} --snps-only --make-bed --out ALL.chr{str(chr_num)}\n",
    "    #Удаляем дубликаты одинаковых SNPs\n",
    "    !cut -f2 ALL.chr{str(chr_num)}.bim | sort | uniq -d > dup{str(chr_num)}.snps\n",
    "    !echo найдено дупликатов в chr{str(chr_num)} хромосоме: \n",
    "    !wc -l dup{str(chr_num)}.snps\n",
    "    !echo ' - удаляю...'\n",
    "    !plink/plink --bfile ALL.chr{str(chr_num)} --exclude dup{str(chr_num)}.snps --make-bed --out ALL.chr{str(chr_num)}                        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 4. Слить полученные файлы в один набор bed\\bim\\fam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Создаем список всех файлов для объединения в один набор\n",
    "#!rm merge_list.txt\n",
    "for i in range(2, 23):\n",
    "    with open('merge_list.txt', 'a') as mlist:\n",
    "        mlist.write(f'ALL.chr{str(i)}.bed ALL.chr{str(i)}.bim ALL.chr{str(i)}.fam \\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ALL.chr2.bed ALL.chr2.bim ALL.chr2.fam \r\n",
      "ALL.chr3.bed ALL.chr3.bim ALL.chr3.fam \r\n",
      "ALL.chr4.bed ALL.chr4.bim ALL.chr4.fam \r\n",
      "ALL.chr5.bed ALL.chr5.bim ALL.chr5.fam \r\n",
      "ALL.chr6.bed ALL.chr6.bim ALL.chr6.fam \r\n",
      "ALL.chr7.bed ALL.chr7.bim ALL.chr7.fam \r\n",
      "ALL.chr8.bed ALL.chr8.bim ALL.chr8.fam \r\n",
      "ALL.chr9.bed ALL.chr9.bim ALL.chr9.fam \r\n",
      "ALL.chr10.bed ALL.chr10.bim ALL.chr10.fam \r\n",
      "ALL.chr11.bed ALL.chr11.bim ALL.chr11.fam \r\n",
      "ALL.chr12.bed ALL.chr12.bim ALL.chr12.fam \r\n",
      "ALL.chr13.bed ALL.chr13.bim ALL.chr13.fam \r\n",
      "ALL.chr14.bed ALL.chr14.bim ALL.chr14.fam \r\n",
      "ALL.chr15.bed ALL.chr15.bim ALL.chr15.fam \r\n",
      "ALL.chr16.bed ALL.chr16.bim ALL.chr16.fam \r\n",
      "ALL.chr17.bed ALL.chr17.bim ALL.chr17.fam \r\n",
      "ALL.chr18.bed ALL.chr18.bim ALL.chr18.fam \r\n",
      "ALL.chr19.bed ALL.chr19.bim ALL.chr19.fam \r\n",
      "ALL.chr20.bed ALL.chr20.bim ALL.chr20.fam \r\n",
      "ALL.chr21.bed ALL.chr21.bim ALL.chr21.fam \r\n",
      "ALL.chr22.bed ALL.chr22.bim ALL.chr22.fam \r\n"
     ]
    }
   ],
   "source": [
    "!cat merge_list.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.merged.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr1\n",
      "  --make-bed\n",
      "  --merge-list merge_list.txt\n",
      "  --out ALL.merged\n",
      "  --threads 4\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: Multiple chromosomes seen for variant 'rs2789253'.\n",
      "Warning: Multiple chromosomes seen for variant 'rs554199249'.\n",
      "Warning: Multiple chromosomes seen for variant '.'.\n",
      "Warning: Multiple chromosomes seen for variant 'rs2843235'.\n",
      "Warning: Unusually long variant ID(s) present.  PLINK 1.9 does not scale well\n",
      "to length-80+ variant IDs; consider using a different naming scheme for long\n",
      "indels and the like.\n",
      "Error: 3 variants with 3+ alleles present.\n",
      "* If you believe this is due to strand inconsistency, try --flip with\n",
      "  ALL.merged-merge.missnp.\n",
      "  (Warning: if this seems to work, strand errors involving SNPs with A/T or C/G\n",
      "  alleles probably remain in your data.  If LD between nearby SNPs is high,\n",
      "  --flip-scan should detect them.)\n",
      "* If you are dealing with genuine multiallelic variants, we recommend exporting\n",
      "  that subset of the data to VCF (via e.g. '--recode vcf'), merging with\n",
      "  another tool/script, and then importing the result; PLINK is not yet suited\n",
      "  to handling them.\n"
     ]
    }
   ],
   "source": [
    "#Пробуем объединить все файлы в один набор\n",
    "!plink/plink --threads 4 --bfile ALL.chr1 --merge-list merge_list.txt --make-bed --out ALL.merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Найдены трёхаллельные варианты\n",
    "Нужно их удалить из файлов\n",
    "немного подредактируем файл ALL.merged-merge.missnp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!rm ALL.merged-merge.missnp\n",
    "variant = ['rs2789253', 'rs2843235', 'rs554199249']\n",
    "for var in variant:\n",
    "    with open('ALL.merged-merge.missnp', 'a') as mlist:\n",
    "        mlist.write(f'{var} \\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ".\r\n"
     ]
    }
   ],
   "source": [
    "!cat ALL.merged-merge.missnp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr1.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr1\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr1\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "6215981 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr1.nosex .\n",
      "--exclude: 6215981 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999953.\n",
      "6215981 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr1.bed + ALL.chr1.bim + ALL.chr1.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr2.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr2\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr2\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "6808662 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr2.nosex .\n",
      "--exclude: 6808662 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999954.\n",
      "6808662 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr2.bed + ALL.chr2.bim + ALL.chr2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr3.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr3\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr3\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "5603253 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr3.nosex .\n",
      "--exclude: 5603253 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999949.\n",
      "5603253 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr3.bed + ALL.chr3.bim + ALL.chr3.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr4.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr4\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr4\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "5500073 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr4.nosex .\n",
      "--exclude: 5500073 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999948.\n",
      "5500073 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr4.bed + ALL.chr4.bim + ALL.chr4.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr5.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr5\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr5\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "5055530 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr5.nosex .\n",
      "--exclude: 5055530 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999953.\n",
      "5055530 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr5.bed + ALL.chr5.bim + ALL.chr5.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr6.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr6\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr6\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "4816865 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr6.nosex .\n",
      "--exclude: 4816865 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999935.\n",
      "4816865 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr6.bed + ALL.chr6.bim + ALL.chr6.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr7.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr7\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr7\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4533180 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr7.nosex .\n",
      "--exclude: 4533180 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99995.\n",
      "4533180 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr7.bed + ALL.chr7.bim + ALL.chr7.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr8.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr8\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr8\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "4434311 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr8.nosex .\n",
      "--exclude: 4434310 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999944.\n",
      "4434310 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr8.bed + ALL.chr8.bim + ALL.chr8.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr9.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr9\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr9\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "3427221 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr9.nosex .\n",
      "--exclude: 3427221 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999948.\n",
      "3427221 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr9.bed + ALL.chr9.bim + ALL.chr9.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr10.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr10\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr10\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "3837170 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr10.nosex .\n",
      "--exclude: 3837170 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999945.\n",
      "3837170 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr10.bed + ALL.chr10.bim + ALL.chr10.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr11.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr11\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr11\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "3891526 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr11.nosex .\n",
      "--exclude: 3891526 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999947.\n",
      "3891526 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr11.bed + ALL.chr11.bim + ALL.chr11.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr12.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr12\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr12\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "3710046 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr12.nosex .\n",
      "--exclude: 3710046 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999954.\n",
      "3710046 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr12.bed + ALL.chr12.bim + ALL.chr12.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr13.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr13\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr13\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2737032 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr13.nosex .\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--exclude: 2737032 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999951.\n",
      "2737032 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr13.bed + ALL.chr13.bim + ALL.chr13.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr14.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr14\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr14\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2548059 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr14.nosex .\n",
      "--exclude: 2548059 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999952.\n",
      "2548059 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr14.bed + ALL.chr14.bim + ALL.chr14.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr15.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr15\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr15\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2328505 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr15.nosex .\n",
      "--exclude: 2328505 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999948.\n",
      "2328505 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr15.bed + ALL.chr15.bim + ALL.chr15.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr16.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr16\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr16\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2607020 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr16.nosex .\n",
      "--exclude: 2607020 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999947.\n",
      "2607020 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr16.bed + ALL.chr16.bim + ALL.chr16.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr17.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr17\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr17\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2234700 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr17.nosex .\n",
      "--exclude: 2234699 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999951.\n",
      "2234699 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr17.bed + ALL.chr17.bim + ALL.chr17.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr18.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr18\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr18\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "2178755 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr18.nosex .\n",
      "--exclude: 2178755 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999949.\n",
      "2178755 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr18.bed + ALL.chr18.bim + ALL.chr18.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr19.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr19\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr19\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "1758439 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr19.nosex .\n",
      "--exclude: 1758439 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999942.\n",
      "1758439 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr19.bed + ALL.chr19.bim + ALL.chr19.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr20.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr20\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr20\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "1745169 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr20.nosex .\n",
      "--exclude: 1745169 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999959.\n",
      "1745169 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr20.bed + ALL.chr20.bim + ALL.chr20.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr21.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr21\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr21\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "1058547 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr21.nosex .\n",
      "--exclude: 1058547 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999928.\n",
      "1058547 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr21.bed + ALL.chr21.bim + ALL.chr21.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.chr22.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr22\n",
      "  --exclude ALL.merged-merge.missnp\n",
      "  --make-bed\n",
      "  --out ALL.chr22\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: --make-bed input and output filenames match.  Appending '~' to input\n",
      "filenames.\n",
      "1059730 variants loaded from .bim file.\n",
      "2504 people (0 males, 0 females, 2504 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to ALL.chr22.nosex .\n",
      "--exclude: 1059730 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 2504 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999946.\n",
      "1059730 variants and 2504 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to ALL.chr22.bed + ALL.chr22.bim + ALL.chr22.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "#Удаляем трёхаллельные варианты\n",
    "for chr_num in range(1, 23):\n",
    "    !plink/plink --bfile ALL.chr{str(chr_num)} --exclude ALL.merged-merge.missnp --make-bed --out ALL.chr{str(chr_num)}    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.15 64-bit (21 Jan 2020)          www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2020 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to ALL.merged.log.\n",
      "Options in effect:\n",
      "  --bfile ALL.chr1\n",
      "  --make-bed\n",
      "  --merge-list merge_list.txt\n",
      "  --out ALL.merged\n",
      "  --threads 4\n",
      "\n",
      "32167 MB RAM detected; reserving 16083 MB for main workspace.\n",
      "Warning: Unusually long variant ID(s) present.  PLINK 1.9 does not scale well\n",
      "to length-80+ variant IDs; consider using a different naming scheme for long\n",
      "indels and the like.\n",
      "\n",
      "Error: Out of memory.  The --memory flag may be helpful.\n",
      "Failed allocation size: 6559540864\n"
     ]
    }
   ],
   "source": [
    "#Заново сливаем все файлы в один набор\n",
    "!plink/plink --threads 4 --memory 24576 --bfile ALL.chr1 --merge-list merge_list.txt --make-bed --out ALL.merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 5. Изменить fam файл, чтобы в нем была информация о популяциях. Соответствие образец-популяция можно найти на сайт 1000 геномов."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Скачиваем таблицу, где можно найти соответствие образец-популяция.\n",
    "!wget 'http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/working/20130606_sample_info/20130606_sample_info.xlsx'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Устанавливаем и импортируем необходимые библиотеки для работы с таблицей\n",
    "#!pip install pandas\n",
    "#!pip install xlrd\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Читаем лист \"Sample info\"\n",
    "df = pd.read_excel('20130606_sample_info.xlsx', sheet_name = 'Sample Info', header = 0, usecols = ['Sample', 'Population', 'Population Description'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Таблица соответствий\n",
    "for pop, desc in zip(df['Population'].unique(), df['Population Description'].unique()):\n",
    "    print (f'{pop} - {desc}')\n",
    "df['Population Description'].unique()\n",
    "df['Population'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Создаем словарь с числовой кодировкой популяций\n",
    "#Европейские кодируем числом больше > 100\n",
    "pop_cod = {}\n",
    "for cnt, samp in enumerate(df.Population.unique()):\n",
    "    if samp in ['GBR', 'FIN', 'IBS', 'TSI']:\n",
    "        pop_cod[samp] = cnt + 3 + 100\n",
    "    else:\n",
    "        pop_cod[samp] = cnt + 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Кодирую популяцию каждого образца в число по ID\n",
    "df['Popnum'] = 0\n",
    "ind_cod = {}\n",
    "for cnt in range(len(df)):\n",
    "    df.iloc[cnt, 3] = pop_cod[df.Population[cnt]]\n",
    "    ind_cod[df.iloc[cnt,0]] = pop_cod[df.Population[cnt]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Считываем fam - файл\n",
    "df_fam = pd.read_csv('ALL.merged.fam', sep = ' ', header = None, index_col = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Меняем обозначение популяции на число из словаря\n",
    "for cnt in range(len(df_fam)):\n",
    "    df_fam.iloc[cnt, 5] = ind_cod[df_fam.iloc[cnt,0]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Перезаписываем fam - файл\n",
    "df_fam.to_csv('ALL.merged.fam', sep = ' ', header = False, index = False, )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 6. Сделать новый набор данных, чтобы в нем были только европейские популяции."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Создаём список европейских популяций"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_keep = df_fam.loc[df_fam[5] > 100][1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "print ('Количество европейских образцов = ', len(df_keep))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_keep.to_csv('keep.fam', sep = ' ', header = False, index = False, )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Делаем набор из европейских популяций"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!plink/plink --bfile ALL.merged --keep-fam keep.fam --make-bed --out ALL.merged.europ"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 7. http://software.genetics.ucla.edu/admixture/ - с помощью этой программы сделать анализ по 5 компонентам (базовый запуск можно найти в документации)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!wget http://software.genetics.ucla.edu/admixture/binaries/admixture_linux-1.3.0.tar.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!tar -xvf admixture_linux-1.3.0.tar.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Запускаем на 4 итерации."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "!adm/admixture -C 4 ALL.merged.europ.bed 5"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
